src/environmentd/tests/sql.rs

```diff
@@ -1321,15 +1381,19 @@ fn insert_with_deterministic_timestamps(
     let mut client_write = server.connect(postgres::NoTls)?;
     let mut client_read = server.connect(postgres::NoTls)?;
 
-    let current_ts = get_explain_timestamp(table, &mut client_read);
+    let mut current_ts = get_explain_timestamp(table, &mut client_read);
     let write_thread = thread::spawn(move || {
         client_write
             .execute(&format!("INSERT INTO {table} VALUES {values}"), &[])
             .unwrap();
     });
-    // Bump `now` by enough for the write to succeed. Table advancements may have increased
-    // the global timestamp by an unknown amount so we bump `now` by a large amount to be safe.
-    *now.lock().expect("lock poisoned") = current_ts + 1_000;
+    while !write_thread.is_finished() {
```

Comment:
Should this just switch out now for SYSTEM_TIME and use ore::Retry to wait? The 1ms sleep here seems fairly busy if the UpdateReadHolds message is only produced once per second.

Reply:
I wanted to avoid increasing `now` by more than necessary and from forcing the caller to switch from a deterministic to the system clock.

This is actually independent of `UpdateReadHolds`. The write has been submitted to the group commit queue and is waiting for `now` to catch up to the system clock. So it doesn't matter what read holds we have. The sleep is just giving group commit a chance to execute. I could have probably left this alone, I just thought that `1_000` was too aggressive and potentially flaky.